lib/assessment.js

```diff
@@ -523,4 +523,44 @@ module.exports = {
       externalGradingSocket.gradingJobStatusUpdated(content.gradingId);
     }
   },
+
+  /**
+   * Updates statistics for all assessments in a course instance, but only if an
+   * update is needed.
+   *
+   * @param {number} course_instance_id - The course instance ID.
+   */
+  async updateAssessmentStatisticsForCourseInstance(course_instance_id) {
+    const result = await sqldb.queryAsync(sql.select_assessments_for_statistics_update, {
+      course_instance_id,
+    });
+    for (const row of result.rows) {
+      await module.exports.updateAssessmentStatistics(row.assessment_id);
+    }
```

Comment:
Might be better to use [`async.eachLimit`](https://caolan.github.io/async/v3/docs.html#eachLimit) since these operations are independent and could potentially run in parallel.

Reply:
Done in https://github.com/PrairieLearn/PrairieLearn/pull/6656/commits/f8d0e83fc17685fa326617f0f777b1c15410034e